pkg/controllers/cloud/networkpolicy_controller.go

```diff
@@ -585,9 +594,28 @@ func (r *NetworkPolicyReconciler) Start(stop context.Context) error {
 	}
 }
 
+// externalEntityStart performs the initialization of the controller.
+// A controller is said to be initialized only when the dependent controllers
+// are synced, and it keeps a count of pending CRs to be reconciled.
+func (r *NetworkPolicyReconciler) externalEntityStart(_ context.Context) error {
+	if err := waitForControllersToSync([]string{CESController}, syncTimeout); err != nil {
```

Comment:
Is there a reason why we do not want to have VMController to the list of controllers to sync?

Reply:
When VMController reconciles an VM CR, it notifies the event to converter module. Converter module calls an API server to check EE CR exists for the VM. If it does not exist, then converter module will create EE CR.

So that is the reason, EE should reconcile all the CRs first, so that converter module does not do anything during restart.

Reply:
Ok got it. So in restart case, for a brief amount of time, there will be only EE CRs without their corresponding VM CRs. That shouldn't cause any issues I suppose.